crates/core/src/host/host_controller.rs

```diff
@@ -46,37 +48,57 @@ type HostCell = Arc<AsyncRwLock<Option<Host>>>;
 /// The registry of all running hosts.
 type Hosts = Arc<Mutex<IntMap<u64, HostCell>>>;
 
-type ExternalStorage = dyn Fn(&Hash) -> anyhow::Result<Option<AnyBytes>> + Send + Sync + 'static;
-type SameDbStorage = dyn Fn(&RelationalDB, &Hash) -> anyhow::Result<Option<AnyBytes>> + Send + Sync + 'static;
+#[async_trait]
```

Comment:
Do we need `async_trait` still?

Reply:
Yes - it lets the future borrow from the ExternalStorage, rather than having to be `'static`